packages/frontend-api/src/Model/Mutation/Customer/User/CustomerUserMutation.php

```diff
@@ -203,4 +203,21 @@ public function editCustomerUserPersonalDataMutation(Argument $argument, InputVa
 
         return $this->customerUserFacade->editCustomerUser($customerUser->getId(), $customerUserData);
     }
+
+    /**
+     * @param \Overblog\GraphQLBundle\Definition\Argument $argument
+     * @return bool
+     */
+    public function removeCustomerUserMutation(Argument $argument): bool
+    {
+        $this->runCheckUserIsLogged();
+
+        $input = $argument['input'];
+
+        $customerUser = $this->customerUserFacade->getByUuid($input['customerUserUuid']);
```

Comment:
Great point. It will fail on error 500. I have found more scenarios, that will end up with 500, e.g `DeliveryAddressFacade::editByCustomer`, there are more, I believe

What about Voter ? How should Voter react on non existing user ? Retun `false` or throw any exception, so it is logged ? 

From my point of view checking existence of entity in queries and mutations (API facades as well) is pain in the ass :roll_eyes: 
Few questions raised: 

* Do we need `*NotFoundUserError` for every entity ? 
* Isn't it enough to return CODE entity-not-found ? 
It would be helpful to implement similar handling like handling `Symfony\Component\HttpKernel\Exception\NotFoundHttpException;` in Symfony requests. Wherever  NotFound exception is raised FE API will throw FeApiNotFoundUserError.

Reply:
*https://github.com/shopsys/shopsys/pull/3286/commits/0a9c67c6d3e7c4cbc1f5d025bec172bd879cb6c2

Reply:
this question should be answered with someone from the frontend department... whether the common code for not found is enough. 
But it seems like a good idea to handle it automatically 👍